salt/metalk8s/archives/unmounted.sls

```diff
@@ -0,0 +1,43 @@
+{# Remove mounted ISOs which were removed from configuration #}
+{%- set desired = salt.metalk8s.get_archives() %}
+{%- set available = salt.metalk8s.get_mounted_archives() %}
+{%- set ns = namespace(found_archives_to_remove=false) %}
+
+{%- for env, archive in available.items() %}
+  {%- if env not in desired %}
+    {%- set ns.found_archives_to_remove = true %}
+
+Unmount '{{ archive.iso }}' from '{{ archive.path }}':
+  mount.unmounted:
+    - name: {{ archive.path }}
+    - device: {{ archive.iso }}
+    - persist: true
+
+Remove archive mountpoint '{{ archive.path }}':
+  file.absent:
+    - name: {{ archive.path }}
+    - unless:
+      - fun: file.path_exists_glob
+        path: {{ archive.path }}/*
+    - require:
+      - mount: Unmount '{{ archive.iso }}' from '{{ archive.path }}'
+
+Mountpoint '{{ archive.path }}' cannot be removed:
```

Comment:
I would prefer a title that does not look like "an error" cause right now even if there is nothing in the directory in the salt output we will have "Mountpoint '{{ archive.path }}' cannot be removed" in the output.
(if `onlyif` is False the state is not executed but you still have it in the output with a comment saying "not executed because of onlyif")

I suggest 
```suggestion
Ensure '{{ archive.path }}' mountpoint is empty
```

Reply:
OK will do that. But to me, it's rather an issue with how the default output shows everything, even when skipped because of requisites.